UnityProject/Assets/Scripts/HealthV2/Living/Metabolism/DigestiveSystemBase.cs

```diff
@@ -0,0 +1,198 @@
+			{
+				if (stomach == null)
+				{
+					stomachList.Remove(stomach);
+					continue;
+				}
+				if (CurrentHunger >= maxHunger) break;
+				CurrentHunger += stomach.ProcessContent();
+			}
+			CurrentHunger = Math.Min(CurrentHunger, maxHunger);
+
+
+			List<BodyFat> fats = new List<BodyFat>(BodyFat); //We might destroy a fat object.
+			foreach (BodyFat fat in fats) //If still not at full hunger, get food from fat stores
+			{
+				if (maxHunger <= CurrentHunger) break;
+
+				CurrentHunger += fat.ConsumeNutrient(maxHunger - CurrentHunger);
+				if (fat.CurrentNutrient <= 0)
+				{
+					BodyFat.Remove(fat);
+					Despawn.ServerSingle(fat.gameObject);
+				}
+			}
+		}
+
+		public void ClampHunger(HungerState hungerState) //Used by Nutrient Pump Implants to prevent hunger from falling below certain values.
+		{
+			//See GetCurrentHungerState for explanantion
+
+			float clampValue = (4 - (int)hungerState) * hungerConsumptionRate;
+
+			CurrentHunger = Math.Max(CurrentHunger, clampValue);
+		}
+
+		public void Initialised(LivingHealthMasterBase livingHealth)
+		{
+			livingHealthMaster = livingHealth;
+			ClampHunger(HungerState.Full);
+
+			var playerHealthV2 = livingHealthMaster as PlayerHealthV2;
+			if (playerHealthV2 != null)
+			{
+				playerHealthV2.PlayerMove.AddModifier(this);
+			}
+			foreach(IStomachProcess stomach in stomachList)
+			{
+				stomach.InitialiseHunger(this);
+			}
+		}
+
+		public void PeriodicUpdate()
+		{
+			InfluenceSpeed();
+		}
+
+		public void AddStomach(IStomachProcess stomach)
+		{
+			stomachList.Add(stomach);
+			maxHunger = CalculateMaxHunger();
+		}
+
+		public void RemoveStomach(IStomachProcess stomach)
+		{
+			stomachList.Remove(stomach);
+			maxHunger = CalculateMaxHunger();
+		}
+
+		public List<IStomachProcess> GetStomachs()
+		{
+			return stomachList;
+		}
+
+		public float CalculateMaxHunger()
+		{
+			float capacity = 0;
+			foreach(IStomachProcess stomach in stomachList)
+			{
+				capacity += stomach.GetStomachMaxHunger();
+			}
+			foreach(BodyFat fat in BodyFat)
+			{
+				capacity += fat.maxNutrientStore;
+			}
+
+			return capacity;
+		}
+
+		private HungerState GetCurrentHungerState()
+		{
+			//If the body needs no food, it makes no sense for it to report being hunger. As such our hunger state relies on our consumption.
+			//Assuming 20 body parts on the average player, with 100 capacity on the human stomach.
+			//This gives us hunger levels of 80 for full, 60 for normal, 40 for hungry, 20 for mal and 0 for starving.
+			//But if we only have a brain, which consumes 1 food, we will need to have less than 5u of food to begin starving.
+
+			float hungerConsump = hungerConsumptionRate == 0 ? hungerConsumptionRate + 0.1f : hungerConsumptionRate; // divide by 0 protect
+			return (HungerState)(Math.Clamp(5 - Math.Ceiling(CurrentHunger / hungerConsump), 0 , 4)); 
+		}
+
+		private void InfluenceSpeed()
+		{
+			var playerHealthV2 = livingHealthMaster as PlayerHealthV2;
```

Comment:
this would be better on per fat body part

Reply:
Why is it a good idea to have a whole bunch of objects fighting for modifiers? Theres now only one hunger value, the value on digestive system. The movement is based off of that value